### PR TITLE
eip7732: add process payload att tests

### PR DESCRIPTION
**Pull request by terencechain**
This PR adds the following payload attestation tests
  - Valid basic attestations (payload present/not present)
  - Partial PTC participation
  - Invalid beacon block root/slot timing
  - Invalid signatures and edge cases

### CODE REVIEW

**Comment by jtraglia** [comment]
You'll need to add `payload_attestation` to the test format too:

* https://github.com/ethereum/consensus-specs/tree/master/tests/formats/operations#condition

File: tests/core/pyspec/eth2spec/test/eip7732/block_processing/test_process_payload_attestation.py [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+
+#
+# Valid payload attestation tests
+#
+
+
+@with_eip7732_and_later
+@spec_state_test
+@always_bls
+def test_process_payload_attestation_valid_basic(spec, state):
```
Comment (jtraglia) [inline tests/core/pyspec/eth2spec/test/eip7732/block_processing/test_process_payload_attestation.py]: Maybe rename to `test_process_payload_attestation_payload_present`.

File: tests/core/pyspec/eth2spec/test/eip7732/block_processing/test_process_payload_attestation.py [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+    spec.process_slots(state, state.slot + 1)
+
+    payload_attestation = prepare_signed_payload_attestation(spec, state, valid_signature=False)
+
+    yield from run_payload_attestation_processing(spec, state, payload_attestation, valid=False)
+
+
+@with_eip7732_and_later
+@spec_state_test
+def test_process_payload_attestation_no_attesting_indices(spec, state):
```
Comment (jtraglia) [inline tests/core/pyspec/eth2spec/test/eip7732/block_processing/test_process_payload_attestation.py]: The second test here is not an invalid signature test. Maybe a new section comment?

Comment (terencechain) [reply tests/core/pyspec/eth2spec/test/eip7732/block_processing/test_process_payload_attestation.py]: Removed the section in [cd2e10e](https://github.com/ethereum/consensus-specs/pull/4463/commits/cd2e10e4c2140a7b75966bbcfc09e22d5cf0ea14)

File: tests/core/pyspec/eth2spec/test/eip7732/block_processing/test_process_payload_attestation.py (hunk 1 above)
Comment (terencechain) [reply tests/core/pyspec/eth2spec/test/eip7732/block_processing/test_process_payload_attestation.py]: Renamed in [cd2e10e](https://github.com/ethereum/consensus-specs/pull/4463/commits/cd2e10e4c2140a7b75966bbcfc09e22d5cf0ea14)

**Review by jtraglia — APPROVED** [review]
LGTM 👍

File: tests/core/pyspec/eth2spec/test/eip7732/block_processing/test_process_payload_attestation.py [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+
+    spec.process_payload_attestation(state, payload_attestation)
+    yield "post", state
+
+
+def get_payload_timeliness_committee(spec, state, slot):
+    """
+    Get the payload timeliness committee for a given slot.
+    """
+    return spec.get_ptc(state, slot)
```
Comment (potuz) [inline tests/core/pyspec/eth2spec/test/eip7732/block_processing/test_process_payload_attestation.py]: Why do we need this helper at all?

File: tests/core/pyspec/eth2spec/test/eip7732/block_processing/test_process_payload_attestation.py [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+    beacon_block_root=None,
+    payload_present=True,
+    attesting_indices=None,
+    valid_signature=True,
+):
+    """
+    Helper to create a signed payload attestation with customizable parameters.
+    """
+    if slot is None:
+        slot = state.slot - 1  # Attest to previous slot
```
Comment (potuz) [inline tests/core/pyspec/eth2spec/test/eip7732/block_processing/test_process_payload_attestation.py:48]: This is pretty bad if `state.slot == 0`

File: tests/core/pyspec/eth2spec/test/eip7732/block_processing/test_process_payload_attestation.py [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+    aggregation_bits = Bitvector[spec.PTC_SIZE]()
+    for i, validator_index in enumerate(ptc):
+        if validator_index in attesting_indices:
+            aggregation_bits[i] = True
+
+    # Create payload attestation data
+    data = spec.PayloadAttestationData(
+        beacon_block_root=beacon_block_root,
+        slot=slot,
+        payload_present=payload_present,
```
Comment (potuz) [inline tests/core/pyspec/eth2spec/test/eip7732/block_processing/test_process_payload_attestation.py:70]: Reminder for later: need to deal with the data bit..

File: tests/core/pyspec/eth2spec/test/eip7732/block_processing/test_process_payload_attestation.py (hunk 4 above)
Comment (terencechain) [reply tests/core/pyspec/eth2spec/test/eip7732/block_processing/test_process_payload_attestation.py:48]: 5f4ce52994b2dd818472050dce95fafdeba0b253

File: tests/core/pyspec/eth2spec/test/eip7732/block_processing/test_process_payload_attestation.py (hunk 3 above)
Comment (terencechain) [reply tests/core/pyspec/eth2spec/test/eip7732/block_processing/test_process_payload_attestation.py]: AI sucks... fixed in 5f4ce52994b2dd818472050dce95fafdeba0b253
